src/modules/roc_core/target_posix/roc_core/secure_random.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2020 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <stdlib.h>
+#include <uv.h>
+
+#include "roc_core/fast_random.h"
+#include "roc_core/panic.h"
+#include "roc_core/secure_random.h"
+
+namespace roc {
+namespace core {
+
+// Based on uv_random from libuv and arc4random_uniform() from OpenBSD.
+uint32_t secure_random(uint32_t from, uint32_t to) {
+#if UV_VERSION_MAJOR >= 1 && UV_VERSION_MINOR >= 33
```

Comment:
When using preprocessor, it's preferred to use it for the entire function instead of a code block inside a function. This approach is less intrusive and easier to read and maintain. For example, [see here](https://www.kernel.org/doc/Documentation/process/coding-style.rst).

Also, don't forget about trailing comments on #else and #endif.

So it should become something like:

```
#if CONDITION
uint32_t secure_random(uint32_t from, uint32_t to) {
    ...
}
#else // !CONDITION
uint32_t secure_random(uint32_t from, uint32_t to) {
    ...
}
#endif // CONDITION
```

Use [this file](https://github.com/roc-project/roc/blob/develop/src/modules/roc_core/target_posixtime/roc_core/time.cpp) as an example.

Reply:
The check `UV_VERSION_MAJOR >= 1 && UV_VERSION_MINOR >= 33` is actually not quite correct. It will not match `2.0` for example.